models/User.php

```diff
@@ -352,6 +353,21 @@ public function attemptConfirmation($code)
     }
 
     /**
+     * Generates a new password and sends it to the user.
+     *
+     * @param string $code Confirmation code.
+     *
+     * @return boolean
+     */
+    public function resendPassword()
+    {
+        $this->password = Password::generate(8);
+        $this->save(true, ['password_hash']);
```

Comment:
Do we need to run validation here? I think it's better to have `false` as the value of the first argument.